server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -229,6 +230,30 @@ final class DefaultSearchContext extends SearchContext {
         }
     }
 
+    private static Executor wrapExecutor(Executor executor) {
+        if (executor instanceof ThreadPoolExecutor tpe) {
```

Comment:
This seems a bit fragile in the event when someone changes our search threadpool in the future to not longer be a ThreadPoolExecutor?

Reply:
I believe we do this in many places in ES, and I don't know how likely it is that we would change the underlying implementation that we rely on. Also, do you have alternatives in mind that we could get the maxPoolSize from? We have at least other two places in search concurrency related code where we rely on this, to get the queue size as well as maxPoolSize to limit the max number of slices we will create.